metal/src/rewrite_rules/rewire_metal_sync.rs

```diff
@@ -25,3 +27,29 @@ pub fn rewire_metal_sync(
         })?;
     Ok(Some(patch))
 }
+
+pub fn rewire_metal_sync_after_const(
+    _ctx: &(),
+    model: &TypedModel,
+    node: &TypedNode,
+    node_name: &str,
+    op: &Const,
+) -> TractResult<Option<TypedModelPatch>> {
+    // Search pattern => Const => ToCPU
+
+    let Some(gpu_const) = op.0.as_metal_tensor() else { return Ok(None) };
+    let cpu_const = gpu_const.to_cpu()?;
+
+    // Identify successor ToCpu
+    let Some(sync_cpu) = next_node(model, node) else { return Ok(None) };
+    let Some(sync_cpu_op) = sync_cpu.op_as::<MetalSync>() else { return Ok(None) };
+    rule_ensure!(sync_cpu_op.kind == MetalSyncKind::ToCpu);
+
+    let mut patch = TypedModelPatch::default();
+
+    let konst_input = patch.taps(model, &node.inputs)?;
+    let out =
+        patch.wire_node(format!("{node_name}"), Const(cpu_const.into(), None), &konst_input)?;
```

Comment:
Mmmm what ? Const takes zero input, so why do we need `konst_input` ?

(Also, if we just want to kill the MetalSync, we can do so with https://docs.rs/tract-core/latest/tract_core/model/struct.ModelPatch.html#method.shunt_one_op)

Reply:
Yes I can remove the tap. However I need to rewrite the Const with the CPU tensor. I can use the replace op also but it is more concise to do a dedicated path ?